devito/function.py

```diff
@@ -400,22 +521,17 @@ def __init__(self, *args, **kwargs):
                 self.time_size = self.time_order + 1
                 self.indices[0].modulo = self.time_size
 
+            self._halo = (0,) + self._halo
```

Comment:
Is the time-halo not the `time_order` as for all other dimensions?

Reply:
not sure; if that were the case, would you not end up writing over a subset of the halo region? I might be wrong...

Reply:
Ahhh ok, this actually shows a deeper problem! The current setup assumes centered differences when it computes the "halo", which is not true for time (after rearranging the primary equation, we usually have a left-sided halo (`stencil=(2, 0)`). The problem here is that we now do not allocate enough entries (we're missing the two "initial condition" read only entries on the left) - or require the user to account for the fact that we magically drop the halo for the time dimension. So this is actually quite a show-stopper.... :thinking:

Reply:
> we now do not allocate

what do you mean by ``now`` ? The ``_halo`` value is computed, but it's basically unused in this PR. I was just laying the foundation for future work